### PR TITLE
fix: contributor card state

### LINKED ISSUE #428: bug: Refreshing the page after closing the 200 contributor guide causes flickering
**Actual Behaviour**

The pop-up window at the bottom right corner of the current home page of the official website will flash when the page is refreshed after being closed manually.

**Expected Behaviour**

There should be no flashing of the refresh page guide window after clicking on the close guide window

**Steps to reproduce it**

1. Click to close 200 contributors Lead
Snip20210727_25

2. Refresh the page
3. 200 Contributor guide component flashes

**Screenshots of the issue**

 

**Would you like to work on the issue?**

### PR DESCRIPTION
**Pull request by Baoyuantop**
Fixes: #428 

Changes:

Screenshots of the change:

### CODE REVIEW

**Comment by Baoyuantop** [comment]
cc @guoqqqi

File: website/src/pages/index.js [hunk 1]
```diff
@@ -259,20 +259,23 @@ const NewsletterSection = () => {
...
-  if (!display || (typeof window !== 'undefined' && localStorage.getItem('SHOW_200_CONTRIBUTOR_EVENT_ENTRY'))) {
-    return null;
-  }
+    if (localStorage.getItem("theme")) {
+      setTheme(localStorage.getItem('theme'));
+    };
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      setTheme('dark');
+    }
+    if (typeof window !== 'undefined' && !localStorage.getItem('SHOW_200_CONTRIBUTOR_EVENT_ENTRY')) {
```
Comment (guoqqqi) [inline website/src/pages/index.js]:
```suggestion
    if (!localStorage.getItem('SHOW_200_CONTRIBUTOR_EVENT_ENTRY')) {
```
In the `useEffect` Hook we can no longer conditionally determine the existence of a window object

File: website/src/pages/index.js [hunk 2]
```diff
@@ -259,20 +259,23 @@ const NewsletterSection = () => {
...
     });
-  }, [theme]);
-
-  if (!display || (typeof window !== 'undefined' && localStorage.getItem('SHOW_200_CONTRIBUTOR_EVENT_ENTRY'))) {
-    return null;
-  }
+    if (localStorage.getItem("theme")) {
+      setTheme(localStorage.getItem('theme'));
+    };
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (guoqqqi) [inline website/src/pages/index.js:272]: Hi, is there more to the description here? I'm not quite sure why we need to listen to themes in this way

Comment (Baoyuantop) [reply website/src/pages/index.js:272]: 1. Switch the operating system to dark mode.
2. open `apisix.apache.org`.
3. website is in dark mode but this component is light.

**Comment by guoqqqi** [comment]
Also, @juzhiyuan @iamayushdas please have a look, when you have time

File: website/src/pages/index.js (hunk 1 above)
Comment (Baoyuantop) [reply website/src/pages/index.js]: Could you please provide a more detailed description of this?

**Review by guoqqqi — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**